### DISCUSSION TITLE
Distinguish fragments from rule calls

### DISCUSSION THREAD

**Discussion by Lotes** [eclipse-langium/langium#567] [Ideas]
It is a minor thing. When you use fragments, they do not distinguish from rule calls. So, when you review a grammar on first sight you think a naked rule call is missing an assignment... but actually it is a fragment call with assignments in it.

My proposal is to either highlight it with a special color, or to have a separate syntax, like `@fragmentName`.

WDYT?

**Comment by montymxb** [comment]
I've wondered a bit about this too. I would think a slight color or syntax change could help, but nothing too great. For reference, here's how a fragment shows up in relation to a parser rule currently. It might also be nice when one hovers over the usage that you could see the associated description of `Fragment` or `ParserRule infers ...`. In such a way you could hover over the components to identify what they are.
<img width="332" alt="Screen Shot 2022-06-28 at 09 15 04" src="https://user-images.githubusercontent.com/5070304/176116969-0318e03a-08ab-489c-b7df-749a76da2aa4.png">

**Comment by spoenemann** [comment]
I wouldn't change the syntax because a parser rule can be without assignment as well (unassigned rule call, see also #205). But using a special color using semantic tokens sounds good.

**Reply by Lotes** [reply]
Added an [issue](https://github.com/langium/langium/issues/602). Sorry for forgetting this thread ^^*
